code/Model/SiteTree.php

```diff
@@ -2940,6 +2970,15 @@ public static function reset()
         }
     }
 
+    /**
+     * Clear the creatableChildren cache on flush
+     */
+    public static function flush()
+    {
+        Injector::inst()->get(CacheInterface::class . '.SiteTree_CreatableChildren')
```

Comment:
This won't respect `$dependencies` config; Could we use `static::singleton()->getCreatableChildrenCache()` instead?